packages/components/fields/select-field/src/select-field.tsx

```diff
@@ -0,0 +1,370 @@
+import {
+  Component,
+  isValidElement,
+  type ReactElement,
+  type ReactNode,
+  type FocusEventHandler,
+} from 'react';
+import {
+  createSequentialId,
+  filterDataAttributes,
+  getFieldId,
+  warning,
+} from '@commercetools-uikit/utils';
+import Constraints from '@commercetools-uikit/constraints';
+import Spacings from '@commercetools-uikit/spacings';
+import FieldLabel from '@commercetools-uikit/field-label';
+import SelectInput from '@commercetools-uikit/select-input';
+import FieldErrors from '@commercetools-uikit/field-errors';
+import type { Props as ReactSelectProps } from 'react-select';
+
+type TErrorRenderer = (key: string, error?: boolean) => ReactNode;
+type TOption = {
+  value: string;
+};
+type TOptionObject = {
+  options: TOption[];
+};
+type TOptions = TOption[] | TOptionObject[];
+type TEvent = {
+  target: unknown;
+};
+
+type TSelectFieldProps = {
+  // SelectField
+  /**
+   * Used as HTML id property. An id is generated automatically when not provided.
+   */
+  id?: string;
+  horizontalConstraint?:
+    | 4
+    | 5
+    | 6
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+  /**
+   * A map of errors. Error messages for known errors are rendered automatically.
+   * <br/>
+   * Unknown errors will be forwarded to renderError.
+   */
+  errors?: {
+    missing?: boolean;
+  };
+  /**
+   * This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.
+   * <br/>
+   */
+  renderError?: TErrorRenderer;
+  /**
+   * Indicates if the value is required. Shows an the "required asterisk" if so.
+   */
+  isRequired?: boolean;
+  /**
+   * Indicates whether the field was touched. Errors will only be shown when the field was touched.
+   */
+  touched?: boolean[] | boolean;
+
+  // SelectInput
+  /**
+   * Aria label (for assistive tech)
+   * <br/>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  'aria-label'?: ReactSelectProps['aria-label'];
+  /**
+   * HTML ID of an element that should be used as the label (for assistive tech)
+   * <br/>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  'aria-labelledby'?: ReactSelectProps['aria-labelledby'];
+  /**
+   * Focus the control when it is mounted
+   */
+  isAutofocussed?: boolean;
+  /**
+   * Remove the currently focused option when the user presses backspace
+   */
+  backspaceRemovesValue?: boolean;
+  /**
+   * Map of components to overwrite the default ones, see what components you can override
+   */
+  components?: {
```

Comment:
Hey @ddouglasz I thought I 'll leave a couple more pesky nitpicks 😅
```suggestion
* * <br/>
   * [Props from React select was used](https://react-select.com/props)
   */
  components?: ReactSelectProps['components'];
```